src/librustc_codegen_ssa/mir/debuginfo.rs

```diff
@@ -341,12 +357,28 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 )
             });
 
-            per_local[var.place.local].push(PerLocalVarDebugInfo {
-                name: var.name,
-                source_info: var.source_info,
-                dbg_var,
-                projection: var.place.projection,
-            });
+            match var.value {
+                mir::VarDebugInfoContents::Place(place) => {
+                    per_local[place.local].push(PerLocalVarDebugInfo {
+                        name: var.name,
+                        source_info: var.source_info,
+                        dbg_var,
+                        projection: place.projection,
+                    });
+                }
+                mir::VarDebugInfoContents::Const(c) => {
+                    let const_val =
+                        if let ConstKind::Value(value) = c.val { value } else { continue };
+                    let operand = OperandRef::from_const(bx, const_val, var_ty);
+
+                    let base =
+                        Self::spill_operand_to_stack(&operand, Some(var.name.to_string()), bx);
+
+                    if let (Some(scope), Some(dbg_var)) = (scope, dbg_var) {
```

Comment:
You can wrap the entire `match` arm in this `if let`, to avoid doing unnecessary work. Although if it got here maybe these are always both `Some` so that might not be relevant.